client/v3/lease.go

```diff
@@ -294,7 +294,9 @@ func (l *lessor) KeepAlive(ctx context.Context, id LeaseID) (<-chan *LeaseKeepAl
 	}
 	l.mu.Unlock()
 
-	go l.keepAliveCtxCloser(ctx, id, ka.donec)
+	if ctx.Done() != nil {
+		go l.keepAliveCtxCloser(ctx, id, ka.donec)
```

Comment:
I think there is not just `ctx.Done()`, but also `ka.donec`. The `l.keepAliveCtxCloser` is also used to watch the event from `ka.donec`.

Reply:
> I think there is not just `ctx.Done()`, but also `ka.donec`. The `l.keepAliveCtxCloser` is also used to watch the event from `ka.donec`.

I think `keepAliveCtxCloser` is only used to watch user's active cancellation. It does watch `ka.donec`, but that is mainly used for exiting the goroutine to avoid leakage if `keepAlive` is closed for other reasons.

Reply:
Thanks for the comment. Change looks good to me.